browser/net/brave_profile_network_delegate.cc

```diff
@@ -82,14 +111,89 @@ BraveProfileNetworkDelegate::BraveProfileNetworkDelegate(
 }
+
+void BraveProfileNetworkDelegate::InitPrefChangeRegistrar() {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  DCHECK(!profile_path().empty());
+
+  // profile path is set on the IO thread so we need to read it here and
+  // then pass the value to the UI thread
+  base::PostTaskWithTraits(
+      FROM_HERE, {BrowserThread::UI},
+      base::Bind(&BraveProfileNetworkDelegate::InitPrefChangeRegistrarOnUI,
+                 base::Unretained(this), profile_path()));
+}
+
+void BraveProfileNetworkDelegate::InitPrefChangeRegistrarOnUI(
+    const base::FilePath& profile_path) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+
+  auto* profile =
+      g_browser_process->profile_manager()->GetProfile(profile_path);
+
+  PrefService* user_prefs = profile->GetPrefs();
+  user_pref_change_registrar_.reset(new PrefChangeRegistrar());
+  user_pref_change_registrar_->Init(user_prefs);
+  user_pref_change_registrar_->Add(
+      kGoogleLoginControlType,
+      base::BindRepeating(&BraveProfileNetworkDelegate::OnPreferenceChanged,
+                          base::Unretained(this),
+                          user_prefs,
+                          kGoogleLoginControlType));
+  user_pref_change_registrar_->Add(
+      kFBEmbedControlType,
+      base::BindRepeating(&BraveProfileNetworkDelegate::OnPreferenceChanged,
+                          base::Unretained(this),
+                          user_prefs, kFBEmbedControlType));
+  user_pref_change_registrar_->Add(
+      kTwitterEmbedControlType,
+      base::BindRepeating(&BraveProfileNetworkDelegate::OnPreferenceChanged,
+                          base::Unretained(this),
+                          user_prefs,
+                          kTwitterEmbedControlType));
+  user_pref_change_registrar_->Add(
+      kLinkedInEmbedControlType,
+      base::BindRepeating(&BraveProfileNetworkDelegate::OnPreferenceChanged,
+                          base::Unretained(this),
+                          user_prefs,
+                          kLinkedInEmbedControlType));
+  UpdateAdBlockFromPref(user_prefs, kFBEmbedControlType);
+  UpdateAdBlockFromPref(user_prefs, kTwitterEmbedControlType);
+  UpdateAdBlockFromPref(user_prefs, kLinkedInEmbedControlType);
+  set_allow_google_auth(user_prefs->GetBoolean(kGoogleLoginControlType));
+}
+
+void BraveProfileNetworkDelegate::OnPreferenceChanged(
+    PrefService* user_prefs,
+    const std::string& pref_name) {
+  UpdateAdBlockFromPref(user_prefs, pref_name);
+}
+
+void BraveProfileNetworkDelegate::UpdateAdBlockFromPref(
+    PrefService* user_prefs,
+    const std::string& pref_name) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  set_allow_google_auth(user_prefs->GetBoolean(kGoogleLoginControlType));
+  std::string tag = GetTagFromPrefName(pref_name);
+  if (tag.length() == 0) {
+    return;
+  }
+  // TODO(bridiver) - user pref can't update global values
+  bool enabled = user_prefs->GetBoolean(pref_name);
+  g_brave_browser_process->ad_block_service()->EnableTag(tag, enabled);
```

Comment:
rather we should listen for pref changes in AdBlock service itself?

Reply:
this code hasn't changed, it just moved from the base class to this class because it doesn't apply to the system delegate